go-controller/pkg/ovn/egressfirewall_test.go

```diff
@@ -396,12 +424,149 @@ var _ = Describe("OVN test basic functions", func() {
 						Port:     400,
 					},
 				},
-				expectedMatch: "(udp && ( udp.dst == 400 )) || (tcp && ( tcp.dst == 100 || tcp.dst == 102 )) || (sctp && ( sctp.dst == 13 ))",
+				expectedMatch: "((udp && ( udp.dst == 400 )) || (tcp && ( tcp.dst == 100 || tcp.dst == 102 )) || (sctp && ( sctp.dst == 13 )))",
 			},
 		}
 		for _, test := range testcases {
 			l4Match := egressGetL4Match(test.ports)
 			Expect(test.expectedMatch).To(Equal(l4Match))
 		}
 	})
+	It("computes correct match function", func() {
+		type testcase struct {
+			ipv4source   string
+			ipv6source   string
+			destinations []matchTarget
+			ports        []egressfirewallapi.EgressFirewallPort
+			output       string
+		}
+		_, cidr, _ := net.ParseCIDR("10.128.0.0/14")
+		config.Default.ClusterSubnets[0].CIDR = cidr
+		testcases := []testcase{
+			{
+				ipv4source:   "testv4",
+				ipv6source:   "",
+				destinations: []matchTarget{{matchKindV4CIDR, "1.2.3.4/32"}},
+				ports:        nil,
+				output:       "match=\"(ip4.dst == 1.2.3.4/32) && ip4.src == $testv4 && ip4.dst != 10.128.0.0/14\"",
```

Comment:
Minor suggestion: use backtick for the string literal instead of quote; then you don't have to escape the quote.

Reply:
done